src/systems.rs

```diff
@@ -60,6 +70,20 @@ pub fn update_action_state<A: Actionlike>(
 
     let mouse = maybe_mouse_input_stream.as_deref();
 
+    if let (Some(input_map), Some(action_state)) =
+        (&mut input_map_resource, &mut action_state_resource)
```

Comment:
This was the kind of formatting I wanted to avoid but oh well.

Reply:
Sorry, but I meant don't use any suffix / prefix at all. In just makes variable longer for no reason.
For example, here In your code `Some(input_map), Some(action_state)` still resources, you just unwrapped the option.

I would rename `input_map_resource` into just `input_map`. In Rust it's okay to shadow varaible names with a different type to avoid having `something_option` or `something_str`.